### ISSUE TITLE
Unable to connect to Flatpak Zotero on Linux

### ISSUE THREAD

**Issue by Victor239** [benjypng/logseq-zoterolocal-plugin#16]
It gives the error message that to please ensure the app is running.

**Comment by Victor239** [comment]
With some further testing using both native Zotero + native Logseq this error still occurs so it's unrelated to Flatpak.

**Comment by benjypng** [comment]
Could you try:

**Interaction with Better BibTex **(https://forums.zotero.org/discussion/114282/zotero-connector-stopped-working-resolved-by-disabling-better-bibtex-add-on)
- If you have this installed, could you disable the plugin temporarily, restart Logseq and Zotero and try again?
- Alternatively, you can try the latest version mentioned  here (https://github.com/retorquere/zotero-better-bibtex/issues/2887) but I think there are some side effects of this approach.

**Testing the local Zotero server** (https://www.zotero.org/support/kb/connector_zotero_unavailable)
- Try to load http://127.0.0.1:23119/connector/ping in the browser.
- Do post the results back here

**Comment by Victor239** [comment]
I've disabled BBT and restarted everything but that has not helped unfortunately. 

Visiting the ping service shows "Zotero is running".

**Comment by Victor239** [comment]
Oh I get a slightly different error in Logseq I think `An unexpected error occurred: Failed to fetch`.

**Comment by benjypng** [comment]
Ok so this means that the green tick in the plugin settings is there correct?
And you are getting this error when you try to do `/Zotero: Insert full item`?

**Comment by Victor239** [comment]
On starting Logseq I get this notification:
![image](https://github.com/user-attachments/assets/cec8e4e6-2476-4a03-b33e-da618f3ea08f)

Going to the plugin settings I see this error:
![image](https://github.com/user-attachments/assets/c7f56a9a-67f6-4b16-8d56-aca0303064ed)

Trying to do `/Zotero: Insert full item` gives this error:
![image](https://github.com/user-attachments/assets/ed78d75d-babe-44b9-980f-a0d60d0c1396)

**Comment by benjypng** [comment]
I've added error messages. Please try the latest version.

**Comment by Victor239** [comment]
I got this error next, which I've resolved in the template page:
![image](https://github.com/user-attachments/assets/9f96640b-8c0a-4ff0-ad4b-4749f3c0d212)

On resolving that and trying to `/Zotero: Insert full item` I then got this error:
![image](https://github.com/user-attachments/assets/d554aa8c-c38f-43e3-bf6d-5e88d7ccb43c)

Activating the Better BibTex plugin in Zotero again hasn't resolved the error. I believe I still have the default settings in Logseq:
![image](https://github.com/user-attachments/assets/877d323b-c7b5-49d0-b364-09a5da8cfff6)

and in Zotero:
![image](https://github.com/user-attachments/assets/72a59071-444b-4346-97be-349616944ff4)

**Comment by benjypng** [comment]
When you do a search using `Zotero: Insert full item`, do you see the item's citeKey in the search results?

**Comment by Victor239** [comment]
No, it says `N/A`. I'm not entirely sure what a cite key even is to be honest, just that it seems to be needed for your plugin.

**Comment by benjypng** [comment]
Oh it actually doesn't! 

If you don't need it, you can actually just do 2 things:
1) In your plugin settings, change your `Page Name Template` to only use `<% title %>`.
2) When you do `/Zotero: Insert template`, remove property that makes reference to `<% citeKey %>`.

**Comment by Victor239** [comment]
Thanks, will try that :) if I do want them though what's wrong with my current setup?

**Comment by benjypng** [comment]
I am not sure what your Zotero set up looks like but it needs to be fixed there. The items not only need to have a citation key created by Better BibTex but the citation key must also be pinned.

**Comment by Victor239** [comment]
Ah I see, I need to manually right-click each Zotero item and pin the key, got it. For now I think I'm fine without BBT and your instructions above worked, thanks for all the help!

**Comment by thelazyoxymoron** [comment]
I also came across the same issue as @Victor239 wrt `citeKey is not configured properly`. Figured out that Better BibTex would automatically pin citation key only after an item was opened, so if you just open the item added in zotero, wait for 1 sec and close that and then retry adding the full item in Logseq, it will work.
